### DISCUSSION TITLE
Why ACCOUNTADMIN and not SYSADMIN

### DISCUSSION THREAD

**Discussion by pyule23** [GoodhandSolutions/terraform-snowflake-playground#24] [General]
Just wondering whether SYSADMIN would be a better role to use as the managed schema owner given it will likely align with all other DB and schema owners within the DB. Perhaps there should be a principle that says ACCOUNTADMIN should not manage any object related matters other than those forced upon us by Snowflake?

**Comment by jagoodhand** [comment]
It is selectable by the user, so actually this is more of a documentation issue than something set by the module. `ACCOUNTADMIN` is only used by the example, if you deployed this with your own Snowflake provider it would deploy using the role you specified in your provider definition.

I'll raise an issue to update the documentation to be more role agnostic though.

Their main reason I would use `ACCOUNTADMIN` is to discourage manual work in the Playground. If it is owned by `ACCOUNTADMIN` then users with SYSADMIN would be forced to use the playground in the same way as other users. Similarly, without `ACCOUNTADMIN` you may run into errors where the automation wouldn't be able to delete some objects in the schema if they were owned by a role 'above' the deploying role.

This does leave the potential for a permission denied error though, which would stop the automation from working. So I'll raise another ticket for a permission denied error to be logged, but not cause the task to fail.

**Reply by jagoodhand** [reply]
Raised as #25 and #26
